app/controllers/application_controller.rb

```diff
@@ -5,8 +5,7 @@ class ApplicationController < ActionController::Base
   before_action :set_locale
 
   def set_locale
-    cookies.permanent[:locale] = params[:locale] if params[:locale].present?
-    I18n.locale = cookies[:locale] || current_user_locale.presence || read_lang_header || I18n.default_locale
+    I18n.locale = params[:locale] || current_user_locale.presence || read_lang_header || I18n.default_locale
```

Comment:
Hi @sillylogger,
 I think we don't need to use `cookie` for this. In case `cookies[:locale]` exist, when user change locale in the profile page, `I18n.locale` will not be changed.

Reply:
@oNguyenNgocTrung I am aware that the `param` / `cookies` would override the preference from the user's locale.  That was my intention to be able to override it.  I am open to changing that intention, but we should address that with a specific commit, not mixed with this address location lookup.